src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +139,966 @@ const ActorInit En_Fr_InitVars = {
+void EnFr_SetupAnimation(EnFr* this, GlobalContext* globalCtx) {
+    EnFr* frog;
+    s32 frogIndex;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B0F8.s")
+    for (frogIndex = 0; frogIndex < ARRAY_COUNT(sEnFrPointers.frogs); frogIndex++) {
+        frog = sEnFrPointers.frogs[frogIndex];
+        if (frog != NULL && frog->actionFunc == EnFr_ChooseJumpFromLogSpot) {
+            continue;
+        } else {
+            return;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B174.s")
+    func_8010BD58(globalCtx, 0x30);
+    this->actionFunc = EnFr_ListeningToOcarinaNotes;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B19C.s")
+void EnFr_ListeningToOcarinaNotes(EnFr* this, GlobalContext* globalCtx) {
+    this->songIndex = FROG_NO_SONG;
+    switch (globalCtx->msgCtx.unk_E3EE) { // Ocarina Song played
+        case 7:
+            this->songIndex = FROG_ZL;
+            break;
+        case 6:
+            this->songIndex = FROG_EPONA;
+            break;
+        case 5:
+            this->songIndex = FROG_SARIA;
+            break;
+        case 8:
+            this->songIndex = FROG_SUNS;
+            break;
+        case 9:
+            this->songIndex = FROG_SOT;
+            break;
+        case 10:
+            this->songIndex = FROG_STORMS;
+            break;
+        case 4:
+            EnFr_OcarinaMistake(this, globalCtx);
+            break;
+        case 1:                                   // Ocarina note played, but no song played
+            switch (globalCtx->msgCtx.unk_E410) { // Jumping frogs in open ocarina based on ocarina note played
+                case FROG_BTN_A:
+                    EnFr_SetupJumpingUp(this, FROG_BLUE);
+                    break;
+                case FROG_BTN_C_DOWN:
+                    EnFr_SetupJumpingUp(this, FROG_YELLOW);
+                    break;
+                case FROG_BTN_C_RIGHT:
+                    EnFr_SetupJumpingUp(this, FROG_RED);
+                    break;
+                case FROG_BTN_C_LEFT:
+                    EnFr_SetupJumpingUp(this, FROG_PURPLE);
+                    break;
+                case FROG_BTN_C_UP:
+                    EnFr_SetupJumpingUp(this, FROG_WHITE);
+                    break;
+            }
+    }
+    if (this->songIndex != FROG_NO_SONG) {
+        this->jumpCounter = 0;
+        this->actionFunc = EnFr_ChildSong;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B22C.s")
+void EnFr_ChildSong(EnFr* this, GlobalContext* globalCtx) {
+    EnFr* frog;
+    u8 songIndex;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B2C8.s")
+    if (this->jumpCounter < 48) {
+        if (this->jumpCounter % 4 == 0) {
+            // 1, 2, 3, 4, 5, 6, 7, 0, 1, 2, 3 (11 celebratory jumps)
+            EnFr_SetupJumpingUp(this, sJumpOrder[(this->jumpCounter >> 2) & 7]); // % 8
```

Comment:
comment probably not needed, someone can look at the data if they want to know the implementation details

Reply:
and if the mod works you should use it. if it doesnt match then you can leave it as & and prob remove the comment